app/models/podcast_planner.rb

```diff
@@ -0,0 +1,152 @@
+class PodcastPlanner
+  attr_accessor :podcast_id, :dates, :selected_days, :week_condition, :period, :monthly_weeks, :start_date, :date_range_condition, :number_of_episodes, :end_date, :publish_time, :segment_count, :drafts
+
+  def initialize(params = {})
+    @dates = []
+    @drafts = []
+    @podcast_id = params[:podcast_id]
+    @start_date = params[:start_date].try(:to_datetime)
+    @selected_days = params[:selected_days].try { reject(&:blank?) }.try { map { |day| day.to_i } }
+    @period = params[:period].try(:to_i)
+    @monthly_weeks = params[:monthly_weeks].try { reject(&:blank?) }.try { map { |week| week.to_i } }
+    @date_range_condition = params[:date_range_condition]
+    @week_condition = params[:week_condition]
+    @number_of_episodes = params[:number_of_episodes].try(:to_i)
+    @end_date = params[:end_date].try(:to_datetime)
+    @publish_time = params[:publish_time].try(:to_time)
+    @segment_count = params[:segment_count].try(:to_i)
+  end
+
+  def date_range_ends_by_episodes?
+    @date_range_condition == "episodes"
+  end
+
+  def date_range_ends_by_end_date?
+    @date_range_condition == "date"
+  end
+
+  def episodes_remain?
+    @dates.length < @number_of_episodes
+  end
+
+  def end_date_passed?(date)
+    date > @end_date
+  end
+
+  def periodic?
+    @week_condition == "periodic"
+  end
+
+  def monthly?
+    @week_condition == "monthly"
+  end
+
+  def ready_to_select_weeks?
+    if periodic?
+      @period.present?
+    elsif monthly?
+      @monthly_weeks.present?
+    else
+      false
+    end
+  end
+
+  def ready_to_select_date_range?
+    if date_range_ends_by_episodes?
+      @number_of_episodes.present?
+    elsif date_range_ends_by_end_date?
+      @end_date.present? && @end_date > @start_date
+    else
+      false
+    end
+  end
+
+  def ready_to_generate_dates?
+    @selected_days.present? &&
+      ready_to_select_weeks? &&
+      ready_to_select_date_range?
+  end
+
+  def generate_dates!
+    return unless ready_to_generate_dates?
+
+    if date_range_ends_by_episodes?
```

Comment:
i also have a bunch of these if/else conditionals because of the different decision trees that could happen from the date generator, and wondered if there was a better way to handle them. at the very least i've tried to abstract and name it as much as possible to provide clarity on where the logic is going.